checkov/terraform/parser2.py

```diff
@@ -0,0 +1,522 @@
+    :param env_vars:                   Optional values to use for resolving environment variables in TF code.
+                                       If nothing is specified, Checkov's local environment will be used.
+    :param module_loader_registry:     Registry used for resolving modules. This allows customization of how
+                                       much resolution is performed (and easier testing) by using a manually
+                                       constructed registry rather than the default.
+    :param dir_filter:                 Determines whether or not a directory should be processed. Returning
+                                       True will allow processing. The argument will be the absolute path of
+                                       the directory.
+    """
+
+    if out_parsing_errors is None:
+        out_parsing_errors = {}
+    if env_vars is None:
+        env_vars = dict(os.environ)
+
+    if include_sub_dirs:
+        for sub_dir, d_names, f_names in os.walk(directory):
+            filter_ignored_directories(d_names)
+            if dir_filter(os.path.abspath(sub_dir)):
+                _internal_dir_load(sub_dir, out_definitions,
+                                   out_evaluations_context, out_parsing_errors, env_vars, None,
+                                   module_loader_registry, dir_filter)
+    else:
+        _internal_dir_load(directory, out_definitions, out_evaluations_context,
+                           out_parsing_errors, env_vars, None, module_loader_registry, dir_filter)
+
+
+def _internal_dir_load(directory: str, out_definitions: Dict,
+                       out_evaluations_context: Dict[str, Dict[str, EvaluationContext]],
+                       out_parsing_errors: Dict[str, Exception],
+                       env_vars: Mapping[str, str],
+                       specified_vars: Optional[Mapping[str, str]],
+                       module_loader_registry: ModuleLoaderRegistry,
+                       dir_filter: Callable[[str], bool]):
+    """
+See `parse_directory` docs.
+    :param specified_vars:     Specifically defined variable values, overriding values from any other source.
+    """
+
+    # Stage 1: Look for applicable files in the directory:
+    #          https://www.terraform.io/docs/configuration/index.html#code-organization
+    #          Load the raw data for non-variable files, but perform no processing other than loading
+    #          variable default values.
+    #          Variable files are also flagged for later processing.
+    var_values: Dict[str, Any] = {}
+    hcl_tfvars: Optional[os.DirEntry] = None
+    json_tfvars: Optional[os.DirEntry] = None
+    auto_vars_files: Optional[List[os.DirEntry]] = None      # lazy creation
+    for file in os.scandir(directory):
+        # Ignore directories and hidden files
+        try:
+            if not file.is_file() or file.name.startswith("."):
```

Comment:
@robeden In this iteration form we don't iterate inner folders. Is this the desired behaviour?

Reply:
It's a little non-obvious because the walk happens in `_parse_directory` (line 84).

I did it this way because Terraform doesn't include subdirectories automatically, so `_internal_dir_load` takes that approach and `_parse_directory` deals with Checkov's recursive loading behavior.

Reply:
@robeden Thanks for the response.
I'm adding some support in loading modules from terraform registry, and I encountered a dilemma: After loading a module, it will only be iterated using `_internal_dir_load` and won't parse any subdirectory. 
What do you think should be done in this context? perhaps replacing the call to `_internal_dir_load` from to `_parse_directory`?

Reply:
I think it should only hit the main directory and then branch off from there based on additional module references. But it’s possible I’m wrong in my understanding. Do you have an example or test case you can point me at?

I should be on slack in an hour or so if you want to talk in real-time there.

Reply:
FYI, @rotemavni and I spoke offline and I think we've convinced ourselves that the current behavior is correct. What we didn't know is that it looks like you can reference deeper into a module with syntax like this ([reference](https://github.com/terraform-aws-modules/terraform-aws-security-group)):
```
module "web_server_sg" {
  source = "terraform-aws-modules/security-group/aws//modules/http-80"
```
So, we'll look for that double slash as the possible entry point.